inferno/tests/test_callbacks.py

```diff
@@ -56,267 +142,76 @@ def test_average_honors_weights(self, avg_loss, history):
+        assert bs_losses == expected
 
-        expected = "Wrong sign 2, expected one of -1, 1."
-        assert exc.value.args[0] == expected
-
-    def test_1_duplicate_key(self, best_loss_cls):
-        key_signs = {'train_loss': 1}
-        with pytest.raises(ValueError) as exc:
-            best_loss_cls(key_signs=key_signs).initialize()
-
-        expected = "BestLoss found duplicate keys: train_loss"
-        assert exc.value.args[0] == expected
-
-    def test_2_duplicate_keys(self, best_loss_cls):
-        key_signs = {'train_loss': 1, 'valid_loss': 1}
-        with pytest.raises(ValueError) as exc:
-            best_loss_cls(key_signs=key_signs).initialize()
-
-        expected = "BestLoss found duplicate keys: train_loss, valid_loss"
-        assert exc.value.args[0] == expected
-
-
-class TestScoring:
-    @pytest.yield_fixture
-    def scoring_cls(self):
-        with patch('inferno.callbacks.to_var') as to_var:
-            to_var.side_effect = lambda x: x
-
-            from inferno.callbacks import Scoring
-            yield partial(
-                Scoring,
-                target_extractor=Mock(side_effect=lambda x: x),
-                pred_extractor=Mock(side_effect=lambda x: x),
-            )
-
-    @pytest.fixture
-    def mse_scoring(self, scoring_cls):
-        return scoring_cls(
-            name='mse',
-            scoring='mean_squared_error',
-        )
-
-    @pytest.fixture
-    def net(self):
-        from inferno.net import History
-
-        net = Mock(infer=Mock(side_effect=lambda x: x))
-        history = History()
-        history.new_epoch()
-        net.history = history
-        return net
-
-    @pytest.fixture
-    def data(self):
-        return [
-            [[3, -2.5], [6, 1.5]],
-            [[1, 0], [0, -1]],
-        ]
-
-    @pytest.fixture
-    def history(self, mse_scoring, net, data):
-        for x, y in data:
-            net.history.new_batch()
-            mse_scoring.on_batch_end(net, x, y, train=False)
-        return net.history
-
-    def test_correct_mse(self, history):
+    def test_correct_mse_values_for_batches(self, history):
         mse = history[:, 'batches', :, 'mse']
-        expected = [[12.5, 1.0]]
+        # for the 4 batches per epoch, the loss is constant
+        expected = [[12.5] * 4, [0.0] * 4, [1.0] * 4]
+        assert np.allclose(mse, expected)
+
+    def test_correct_mse_values_for_epoch(self, history):
+        mse = history[:, 'mse']
+        expected = [12.5, 0.0, 1.0]
         assert np.allclose(mse, expected)
 
+    def test_correct_mse_is_best(self, history):
+        is_best = history[:, 'mse_best']
+        assert is_best == [True, True, False]
+
     def test_other_score_and_name(self, scoring_cls, net):
+        """Test that we can change the scoring to accuracy score."""
         scoring = scoring_cls(
             name='acc',
             scoring='accuracy_score',
         )
         for x, y in zip(np.arange(5), reversed(np.arange(5))):
```

Comment:
I think this warrants a comment what this x/y data is expected to give as a result. I know that the net is just the identity function but only because I read the whole test code. This is not clear from just looking at this test.